core/src/main/java/org/infinispan/util/CacheInjectable.java

```diff
@@ -0,0 +1,15 @@
+package org.infinispan.util;
+
+import org.infinispan.Cache;
+
+/**
+ * Interface that describes how a cache can be injected into another object.  This is useful for cases such as
+ * after an object is deserialized and you must inject a Cache into it.
+ */
+public interface CacheInjectable<K, V> {
```

Comment:
Interface name sounds a bit odd to me, how about `CacheAware`?

Also, `org.infinispan.util` looks like an implementation package to me, even if it's not marked as such... I think this would go better in `org.infinispan.stream`.